kvmagent/kvmagent/test/test_ha_fencer.py

```
@@ -1,42 +1,237 @@
+                    if path and path.startswith("/dev/"):
+                        self.volumes.append(path)
+            if is_root_vol:
+                self.root_volume = path
 
-class TestAbstractStorageFencer(unittest.TestCase):
-    @mock.patch('kvmagent.plugins.ha_plugin.AbstractStorageFencer.read_fencer_hearbeat')
-    def test_check_fencer_heartbeat_raises_exception(self, mock_read_fencer_heartbeat):
-        # Arrange
-        fencer = FakeStorageFencer()
+        self.xml = xml
+        root = etree.fromstring(xml)
```

Ruff (0.15.10)

[error] Using `xml` to parse untrusted data is known to be vulnerable to XML attacks; use `defusedxml` equivalents

(S314)